torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8333,6 +8333,24 @@ def _sample_rightmost_arg(self, opinfo, rightmost_arg_type, device, dtype, num_t
                 requires_grad=_foreach_inputs_kwargs.get("requires_grad", False),
             )]
         should_use_simpler_scalars = opinfo.name == "_foreach_pow" and dtype in (torch.float16, torch.bfloat16)
+        # passing float scalar with int tensors leads to unsafe cast
+        should_use_int_scalars = opinfo.name == "_foreach_addcmul" and dtype in integral_types()
+        # same for complex
+        should_use_int_float_scalars = (
+            opinfo.name == "_foreach_addcmul" and dtype in floating_types_and(torch.float16, torch.bfloat16)
+        )
```

Comment:
This is needed to generate samples for foreach_addcmul that don't cause errors due to unsafe casts.